### PR TITLE
Our Client supports PKCE-S256

### PR DESCRIPTION
**Pull request by IljaN**
## Description
Override parent getCodeChallengeMethod to signify that our client knows PKCE-S256

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: lib/Client.php [hunk 1]
```diff
@@ -181,6 +181,13 @@ protected function fetchURL($url, $post_body = null, $headers = []) {
 		return parent::fetchURL($url, $post_body, $headers);
 	}
 
+	/**
+	 * @codeCoverageIgnore
+	 */
+	public function getCodeChallengeMethod() {
+		return 'S256';
```
Comment (DeepDiver1975) [inline lib/Client.php:188]: has to be a config parameter - this is not true for all IdPs on this planet

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

File: lib/Client.php (hunk 1 above)
Comment (IljaN) [reply lib/Client.php:188]: @DeepDiver1975 This is as far as I understood to signify which challenge methods the Client supports, the IDP capability is always read from the .well-known. 

See: https://github.com/jumbojett/OpenID-Connect-PHP/blob/master/src/OpenIDConnectClient.php#L664

Comment (IljaN) [reply lib/Client.php:188]: Also See: https://github.com/jumbojett/OpenID-Connect-PHP/blob/master/src/OpenIDConnectClient.php#L1812

I supose it is built like that to accomodate for implementations which extend from the jumbojett base client.

Comment (DeepDiver1975) [reply lib/Client.php:188]: You are right! I should have read the code base more in depth!

**Review by DeepDiver1975 — APPROVED** [review]

**Review by DeepDiver1975 — APPROVED** [review]

**Pull request merged**